### PR TITLE
Display weather at geographical midpoint of the team

### PR DESCRIPTION
**Pull request by 1campbellj**
Display an additional card that represents the geographical midpoint of all team members.

Given the members, calculate the midpoint lat/lon from existing lat/lons. Use reverse lookup to identify a city name near that location. Then look up the weather at that location and attach it to a fake member that lives at that location.
Screen Shot 2022-05-21 at 2 53 23 AM
Screen Shot 2022-05-21 at 2 35 48 AM
Screen Shot 2022-05-21 at 2 35 36 AM

### CODE REVIEW

File: app/routes/teams/$id.tsx [hunk 1]
```diff
@@ -98,6 +99,45 @@ function WeatherCard({ weather }: { weather: any }) {
...
+function MidpointItem({ team }: { team: Team }) {
+  const fetcher = useFetcher();
+  React.useEffect(() => {
+    if (fetcher.type === "init") {
+      fetcher.load(`/teams/${team.id}/midpoint_weather`);
+    }
+  }, [fetcher]);
+
+  return (
+    <li key={'midpoint-item'}>
```
Comment (tjefferson08) [inline app/routes/teams/$id.tsx]: unnecessary interp & mixed quotes here (run prettier?)

File: app/routes/teams/$team_id.midpoint_weather.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+  const openWeatherURL = new URL(
+    `https://api.openweathermap.org/data/2.5/weather`
+  );
+  openWeatherURL.searchParams.set("units", "imperial");
+  openWeatherURL.searchParams.set("lat", String(midpoint.lat));
+  openWeatherURL.searchParams.set("lon", String(midpoint.lon));
+  openWeatherURL.searchParams.set("appid", context.env.OPEN_WEATHER_API_KEY);
+  console.log("fetching", openWeatherURL.toString());
+  const weather = (await (
```
Comment (tjefferson08) [inline app/routes/teams/$team_id.midpoint_weather.tsx]: I think you can run this API request and the previous one in ||

File: app/routes/teams/$team_id.midpoint_weather.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    await fetch(openWeatherURL.toString(), {
+      headers: { Accept: "application/json" },
+    })
+  ).json()) as any[];
+
+  const fakeMember = {
+    name: "Midpoint Mertle",
```
Comment (tjefferson08) [inline app/routes/teams/$team_id.midpoint_weather.tsx:45]: :joy:

File: app/routes/teams/$team_id.midpoint_weather.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  };
+
+  return json(
+    { ...fakeMember, weather },
+    { headers: { "Cache-Control": "max-age=300, s-maxage=3600" } }
+  );
+};
+
+// Formula taken from http://www.geomidpoint.com/calculation.html
+function midPoint(geos: Array<Point>) {
```
Comment (tjefferson08) [inline app/routes/teams/$team_id.midpoint_weather.tsx:58]: :exploding_head:

**Review by tjefferson08 — COMMENTED** [review]
Awesome! Very fun & whimsical idea

You want this feature deployed or was this entirely for educational purposes?

**Comment by 1campbellj** [comment]
> Awesome! Very fun & whimsical idea
> 
> You want this feature deployed or was this entirely for educational purposes?

It's your app! Merge it if you like, if you'd rather not merge it I'm cool with it just being a learning exercise

File: app/routes/teams/$id.tsx (hunk 1 above)
Comment (1campbellj) [reply app/routes/teams/$id.tsx]: Yeah I'll see if I can get prettier up and going on nvim

File: app/routes/teams/$team_id.midpoint_weather.tsx (hunk 2 above)
Comment (1campbellj) [reply app/routes/teams/$team_id.midpoint_weather.tsx]: Is this what you meant? Using `Promise.all`?

**Pull request merged**